api-gen/extraction/index.ts

```diff
@@ -0,0 +1,11 @@
+import {writeFileSync} from 'fs';
+
+function main() {
+  const [srcs, outputFileName] = process.argv.slice(2);
```

Comment:
Optional Nit/For a follow-up, or here: I generally like to use the term `outputExecrootRelativePath` or something as this just makes it clear that this a execroot-relative path that is expected to be passed to the action (I know it's long-but just explicit)